ktor-shared/ktor-junit/jvm/src/io/ktor/junit/Assertions.kt

```diff
@@ -29,3 +31,10 @@ fun <T> assertAll(collection: Iterable<T>, assertion: (T) -> Unit) {
         }
     )
 }
+
+inline fun <reified T : Any> assertSerializable(obj: T): T {
+    val encoded = ByteArrayOutputStream().also {
+        ObjectOutputStream(it).writeObject(obj)
+    }.toByteArray()
+    return ObjectInputStream(encoded.inputStream()).readObject() as T
```

Comment:
Should the `assertEquals` call be right here?
```suggestion
    val decoded = ObjectInputStream(encoded.inputStream()).readObject() as T
    assertEquals(obj, decoded)
    return decoded
```

Reply:
It would have to be optional (default true) because not all objects might be equal after deserialization, though in most cases they would be. But yes, that's actually how we do in our implementation. I was just afraid of the code looking 100% the same...